### PR TITLE
Test to fix upgrade from noarch file

### PR DESCRIPTION
**Pull request by inknos**
https://github.com/rpm-software-management/dnf/pull/1840

### CODE REVIEW

File: dnf-behave-tests/dnf/upgrade-from-noarch.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Background: Install package to upgrade
+   Given I use repository "upgrade-from-noarch"
+    When I execute dnf with args "install it-1.0"
+    Then the exit code is 0
+     And DNF Transaction is following
+         | Action        | Package         |
+         | install       | it-1.0-1.x86_64 |
+
+Scenario: Upgrade to rpm from file which is not noarch
+   Given I execute dnf with args "install it.noarch --downloadonly --downloaddir={context.dnf.installroot}/it"
```
Comment (m-blaha) [inline dnf-behave-tests/dnf/upgrade-from-noarch.feature:12]: "Given I **successfully** execute..."?
Or better, you do not need to download the package, you can directly use fixtures path in the following `I execute dnf with args` step: `{context.dnf.fixturesdir}/repos/upgrade-from-noarch/noarch/it-2.0-1.noarch.rpm`

File: dnf-behave-tests/dnf/upgrade-from-noarch.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    When I execute dnf with args "install {context.dnf.installroot}/it/it-2.0-1.noarch.rpm"
```
Comment (m-blaha) [inline dnf-behave-tests/dnf/upgrade-from-noarch.feature:13]: The test passes even on unpatched dnf. I guess you should use the `upgrade` command here, `install` command is not affected by the bug.

**Review by m-blaha — CHANGES_REQUESTED** [review]

File: dnf-behave-tests/dnf/upgrade-from-noarch.feature (hunk 1 above)
Comment (inknos) [reply dnf-behave-tests/dnf/upgrade-from-noarch.feature:12]: thanks, I fixed it

File: dnf-behave-tests/dnf/upgrade-from-noarch.feature (hunk 2 above)
Comment (inknos) [reply dnf-behave-tests/dnf/upgrade-from-noarch.feature:13]: I am testing again, good catch

Comment (inknos) [reply dnf-behave-tests/dnf/upgrade-from-noarch.feature]: @m-blaha should be good to go now

**Pull request merged**